### DISCUSSION TITLE
Realm

### DISCUSSION THREAD

**Discussion by mahirooo** [lit-iphone-course/lit-iphone-course#66] [Question-original] [answered]
(1) 実現したいこと
stampImageで保存したものを、ViewController上のそれぞれのCollectionViewに反映させるために、Realm上にstampImage配列を”**別々に**”保存したい

機能的にはPushViewControllerで長押しするとスタンプが押され、ViewControllerに戻るとそれが反映されるようにしたい。

(2) 実現するために自分が試した内容とその結果
RealmにstampImage配列をListに変換して保存することはできた。
ただ一つのchildにstampImage配列の中身が追加され続けてしまい、別々に保存することができない

(3) 調べた内容や資料
https://qiita.com/rei_012/items/df430d228577c041d61d
https://qiita.com/dieKn/items/7df1f7992f9f04661c51

(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)

```
class Parent: Object {
    let children = List<Child>()
}
class Child: Object {
    @objc dynamic var title: String = ""
}
```

```
@IBAction func save(){
        let parent = Parent()
        let children = parent.children
        for img in stampImage {
            let child = Child()
            child.title = img
            children.append(child)
        }
        try! realm.write() {
            realm.add(parent)
        }
        
        let alert: UIAlertController = UIAlertController(title: "成功",message: "保存しました",preferredStyle: .alert)

        alert.addAction(
            UIAlertAction(title: "OK", style: .default) { [weak self] _ in
                self?.navigationController?.popViewController(animated: true)
            }
        )
        print("保存したよ")
    }

```
(5) GitHubのリンク
https://github.com/mahirooo/TampStamp

**Comment by yukiny0811** [comment, marked as answer]
あ！別々にってそういうことか！
単純にParentを複数作ればOKだよ！
```.swift
let parents = realm.objects(Parent.self)
```
ってやると、このparentsの中身は、
```.swift
parents = [
    Parent(children: ["aaa", "bbb", "ccc"]),
    Parent(children: ["kkk, "lll", "sss"]),
    Parent(children: ["iii", "www", "aaa"])
]
```
こんな感じになる！

**Reply by yukiny0811** [reply]
こんな感じで！
```.swift
let parent1 = Parent()
let parent2 = Parent()
try! realm.write() {
    realm.add(parent1)
    realm.add(parent2)
}
```

**Reply by mahirooo** [reply]
なるほど！
このアプリの機能的に
①スタンプカードを作る
②ViewControllerのTableViewにカードが反映される
③スタンプカードCellをクリックしてPushViewControllerに移動
④スタンプを長押しで押して内容を保存
⑤ViewControllerに戻るとスタンプの内容が反映されてる
⑥ 3〜5を繰り返す

って流れになるんだけど、カードごとにスタンプの数(stampImage配列)を管理する＝Parentをカードごとに作成するってことだと思っていて、そうなると②のスタンプカードを作ったときにlet parent = Parent()が個別で作られるようにしなきゃいけない気がしてるんだけどそれって合ってる？？&それってできる......？

**Reply by yukiny0811** [reply]
できる！
(zoomで解決済み)
